crates/nu-protocol/src/engine/engine_state.rs

```diff
@@ -323,27 +323,39 @@ impl EngineState {
             .any(|(overlay_name, _)| name == overlay_name)
     }
 
-    pub fn active_overlay_ids(&self, removed_overlays: &[Vec<u8>]) -> Vec<OverlayId> {
+    pub fn active_overlay_ids<'a, 'b>(
+        &'b self,
+        removed_overlays: &'a [Vec<u8>],
+    ) -> impl DoubleEndedIterator<Item = &OverlayId> + 'a
```

Comment:
Are we maybe using the wrong data structure when in practice we basically always call `.rev()` outside and iterate regularly inside `EngineState` and `ScopeFrame`?

I know that the contiguous access should be relatively fine from both directions so maybe it is not really a performance concern and more an observation how we use it and may accidentally cause a bug based on direction.

Reply:
@kubouch thoughts?